src/main/clojure/clara/tools/tracing.cljc

```diff
@@ -118,4 +118,4 @@
                          (filter #(instance? PersistentTracingListener %) )
                          (first))]
     (.-trace ^PersistentTracingListener listener)
-    (throw (IllegalStateException. "No tracing listener attached to session."))))
+    (throw (ex-info "No tracing listener attached to session." {}))))
```

Comment:
Minor: it would be better to have the session in the data map.  i.e. 

````
(throw (ex-info "No tracing listener attached to session." {:session session}))
````
It is nice to have the offending object in the exception so that it can be examined in the REPL when an exception is thrown during development.